### PR TITLE
Web console: polish the data loader

### PR DESCRIPTION
**Pull request by vogievetsky**
Several little things to make the data loader work better:

image

- Rearrange the fields in the Tuning step to be in a better order
- Make `useEarliestOffset` required in Kafka ingestion
- Make `segmentGranularity` required in batch ingestion 
- Fix bug in numeric input where you could not erase all chars if there was a `defaultValue` set

### CODE REVIEW

**Comment by ccaominh** [comment]
Travis failure looks unrelated (i.e., flaky integration test) and needs to be retriggered: https://travis-ci.org/apache/incubator-druid/jobs/586380336

**Comment by fjy** [comment]
@vogievetsky can we make sure the default for useEarliestOffset is set to whatever the default is in Druid? I think it is false

**Comment by vogievetsky** [comment]
@fjy this PR intentionally removes the default value from `useEarliestOffset`. It makes it required instead. Look at the screenshot I posted, after this PR you can no longer just click "Next" past the Tuning step. This is because it is very important for the user to consider `useEarliestOffset` and set it accordingly instead of just going with the default.

Since the default is `false` right now if you run through the Kafka quickstart and do not read the instructions and just click `Next`, `Next`, `Next` (something that is very tempting to do) you will end up in a bad state where no data is coming in because no new data is flowing into Kafka.

I have done the same for the `segmentGranularity` for batch ingestion.

TL;DR: think of this PR as data loader speed humps intentionally there to flow the user down a bit in strategic places.

**Comment by vogievetsky** [comment]
I should add that I have personally stepped on the two rakes I described in the comment above.

1. Use `useEarliestOffset: false (default)` in a static Kafka topic
2. Use `segmentGranularity: DAY` in a super sparse multi-year data source

I can not be trusted with the `Next` button!
